students/BrianNisonger/Lesson9/jpgdiscover.py

```diff
@@ -0,0 +1,30 @@
+"""
+Recursively go through directories looking for pngs
+"""
+import os
+
+
+def jpegdiscovery(directory):
+    """
+    Recursively adds files to a dict if a png, define dictionary before calling
+    """
+    files = os.listdir(directory)
+    for file in files:
+        file_path = os.path.join(directory, file)
+        if os.path.isdir(file_path):
+            jpegdiscovery(file_path)
+        elif file.endswith(".png"):
+            if directory in file_dict.keys():
+                file_dict[directory].append(file)
+            else:
+                file_dict[directory] = [file]
+
+    return [[file, file_dict[file]] for file in file_dict]
```

Comment:
Not sure that it is really cleaner, if the user has to instantiate an empty dictionary outside of your function. I think even with the dictionary, if you had file_dict as a keyword argument, assigned to None, like we did in class, you wouldn't need to have it in the outside scope.